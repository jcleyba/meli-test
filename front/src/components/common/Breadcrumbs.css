.breadcrumbs {
  list-style: none;
  display: flex;
  padding: 0;
  margin: 16px auto;
  font-size: 12px;
  color: #999; }
  .breadcrumbs .breadcrumb {
    display: flex;
    align-items: center; }
    .breadcrumbs .breadcrumb:last-child .chevron {
      display: none; }
  .breadcrumbs .chevron {
    margin: 0 2px;
    display: flex;
    align-items: center; }
    .breadcrumbs .chevron img {
      width: 10px; }
