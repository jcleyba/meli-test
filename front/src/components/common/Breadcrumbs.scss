@import '../../styles/_variables';

.breadcrumbs {
  list-style: none;
  display: flex;
  padding: 0;
  margin: $box_padding auto;
  font-size: $location_font_size;
  color: $grey;
  .breadcrumb {
    display: flex;
    align-items: center;

    &:last-child {
      .chevron {
        display: none;
      }
    }
  }
  .chevron {
    margin: 0 2px;
    display: flex;
    align-items: center;

    img {
      width: 10px;
    }
  }
}
