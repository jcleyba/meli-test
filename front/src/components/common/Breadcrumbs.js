import React from 'react';

import './Breadcrumbs.css';
import chevron from '../../images/chevron-right.svg';

const Breadcrumbs = props => {

  const renderBreadcrumbs = list => {
    const breadcrumbs = list.map((item, index) => {
      return (
        <li key={index} className="breadcrumb">
          <div className="name">{item.name}</div>
          <div className="chevron">
            <img src={chevron} alt="chevron"/>
          </div>
        </li>
      )
    });

    return <ol className="breadcrumbs">{breadcrumbs}</ol>
  };
  return renderBreadcrumbs(props.pathFromRoot);
};

export default Breadcrumbs;
