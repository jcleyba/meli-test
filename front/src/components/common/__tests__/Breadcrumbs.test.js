import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-enzyme';

import Breadcrumbs from '../Breadcrumbs';

describe('Breadcrumbs Component', () => {

  let wrapper;
  let mounted;

  const props = {
    pathFromRoot: [{
      id: 1,
      name: 'Name',
    }],
  };


  beforeEach(() => {
    wrapper = shallow(<Breadcrumbs {...props}/>);
    mounted = mount(<Breadcrumbs {...props}/>);
  });

  afterEach(() => {
    wrapper = null;
    mounted = null;
  });

  describe('Breadcrumbs', () => {

    it('matches current snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have pathFromRoot prop', () => {
      expect(mounted).toHaveProp('pathFromRoot');
    });

    it('should have a breadcrumbs className', () => {
      expect(mounted.find('.breadcrumbs')).toBePresent();
    });

    it('should render an ordered list', () => {
      expect(mounted.find('ol')).toBePresent();
    });

    it('should render at least one li item', () => {
      expect(mounted.find('li')).toBePresent();
    });

    it('should have a breadcrumb on li items className', () => {
      expect(mounted.find('li.breadcrumb')).toBePresent();
    });

    it('should render at least one chevron item', () => {
      expect(mounted.find('.chevron img')).toBePresent();
    });

    it('should render the name of pathFromRoot array', () => {
      expect(mounted.find('.name').text()).toEqual(props.pathFromRoot[0].name)
    });
  });
})
;
