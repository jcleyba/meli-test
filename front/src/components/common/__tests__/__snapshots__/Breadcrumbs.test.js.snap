// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Component Breadcrumbs matches current snapshot 1`] = `"<ol class=\\"breadcrumbs\\"><li class=\\"breadcrumb\\"><div class=\\"name\\">Name</div><div class=\\"chevron\\"><img src=\\"chevron-right.svg\\" alt=\\"chevron\\"/></div></li></ol>"`;
